interceptor.go

```diff
@@ -270,56 +221,100 @@ func (i *interceptor) WrapStreamingHandler(next connect.StreamingHandlerFunc) co
 			}
 		}
 		state := streamingState{
-			protocol: protocolToSemConv(req.Peer),
-			attrs:    attributesFromRequest(req),
+			protocol:   protocolToSemConv(req.Peer.Protocol),
+			attributes: requestAttributes(req),
 		}
+		carrier := propagation.HeaderCarrier(conn.RequestHeader())
+		name := strings.TrimLeft(conn.Spec().Procedure, "/")
+		spanCtx := trace.SpanContextFromContext(ctx)
+		ctx = i.config.propagator.Extract(ctx, carrier)
+		ctx, span := i.config.tracer.Start(
+			trace.ContextWithRemoteSpanContext(ctx, spanCtx),
+			name,
+			trace.WithSpanKind(trace.SpanKindServer),
+			trace.WithAttributes(state.attributes...),
+		)
+		defer span.End()
+		i.config.propagator.Inject(ctx, carrier)
 		streamingHandler := &streamingHandlerInterceptor{
 			StreamingHandlerConn: conn,
 			receive: func(msg any, conn connect.StreamingHandlerConn) error {
-				return state.receive(ctx, instr, msg, conn)
+				size, err := state.receive(msg, conn)
+				if errors.Is(err, io.EOF) {
+					return err
+				}
```

Comment:
Another issue currently is on io.EOF events span events and metrics with 0 size will be recorded when they really shouldn't be